service/models.py

```diff
@@ -0,0 +1,371 @@
+        :type customer: int
+
+        :return: a collection of Wishlists in that customer
+        :rtype: list
+
+        """
+        logger.info("Processing customer query for %s ...", customer)
+        return cls.query.filter(cls.customer == customer)
+
+class Item(db.Model):
+    """
+    Class that represents a Item
+
+    This version uses a relational database for persistence which is hidden
+    from us by SQLAlchemy's object relational mappings (ORM)
+    """
+
+    __tablename__ = 'item'
+
+    app = None
+
+    ##################################################
+    # Table Schema
+    ##################################################
+    id = db.Column(db.Integer, primary_key=True)
+    name = db.Column(db.String(63), nullable=False)
+    wishlist_id = db.Column(db.Integer, db.ForeignKey('wishlist.id'))
+    wish2item = db.relationship('Wishlist', backref='wish2item')
+
+
+    ##################################################
+    # INSTANCE METHODS
+    ##################################################
+
+    def __repr__(self):
+        return "<Item %r id=[%s]>" % (self.name, self.id)
+
+    def create(self, wishlist_id):
+        """
+        Creates a Item to the database
+        """
+        logger.info("Creating %s", self.name)
+        self.id = None  # id must be none to generate next primary key
+        self.wishlist_id = wishlist_id
+        db.session.add(self)
+        db.session.commit()
+
+    def update(self):
+        """
+        Updates a Item to the database
+        """
+        logger.info("Saving %s", self.name)
+        if not self.id:
+            raise DataValidationError("Update called with empty ID field")
+        db.session.commit()
+
+    def delete(self):
+        """Removes a Item from the data store"""
+        logger.info("Deleting %s", self.name)
+        db.session.delete(self)
+        db.session.commit()
+
+    def serialize(self):
+        """Serializes a Item into a dictionary"""
+        return {
+            "id": self.id,
+            "name": self.name,
+            "wishlist_id": self.wishlist_id,
+        }
+
+    def deserialize(self, data):
+        """
+        Deserializes a Item from a dictionary
+
+        :param data: a dictionary of attributes
+        :type data: dict
+
+        :return: a reference to self
+        :rtype: Item
+
+        """
+        try:
+            self.name = data["name"]
+            self.wishlist_id = data["wishlist_id"]
+        except KeyError as error:
+            raise DataValidationError("Invalid item: missing " + error.args[0])
+        except TypeError as error:
+            raise DataValidationError(
+                "Invalid item: body of request contained bad or no data"
+            )
+        return self
+
+    ##################################################
+    # CLASS METHODS
+    ##################################################
+
+    @classmethod
+    def init_db(cls, app):
```

Comment:
it seems a little bizarre that this is defined on each model, but i see it is in the template so i guess let's go with it 😅